pkgs/applications/networking/p2p/transgui/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv, fetchsvn, pkgconfig, makeDesktopItem, unzip, fpc, lazarus,
+libX11, glib, gtk, gdk_pixbuf, pango, atk, cairo }:
+
+stdenv.mkDerivation rec {
+  name = "transgui-trunk-r${version}";
+  version = "986";
+  
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/transgui/code/trunk/";
+    rev = "${version}";
+    sha256 = "0z83hvlhllm6p1z4gkcfi1x3akgn2xkssnfhwp74qynb0n5362pi";
+  };
+
+  buildInputs = [ pkgconfig unzip fpc lazarus stdenv.cc libX11
+                  glib gtk gdk_pixbuf pango atk cairo
+                ];
+                
+  makeFlags = [
+    "FPC=fpc"
+    "PP=fpc"
+    "INSTALL_PREFIX=$(out)"
+    "LCL_PLATFORM=gtk2"
+  ];
+
+ desktopItem = makeDesktopItem rec {
+    name = "transgui";
+    exec = name;
+    icon = name;
+    comment = meta.description;
+    desktopName = "Transmission Remote GUI";
+    genericName = "BitTorrent Client";
+    categories = "Application;Network;FileTransfer;P2P;GTK;";
+    mimeType = stdenv.lib.concatStringsSep ";" [
+      "application/x-bittorrent" "x-scheme-handler/magnet"
+    ];
+  };
+  
+  preBuild = ''
+    export makeFlags="$makeFlags INSTALL_PREFIX=$out/"
+    export NIX_LDFLAGS="$NIX_LDFLAGS -L${stdenv.cc.cc.lib}/lib \
+                        -lX11 -lglib-2.0 -lgtk-x11-2.0 -lgdk-x11-2.0 \
+                        -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo \
+                        -lc"
+    export LCL_PLATFORM=gtk2
+  '';
+
+  postInstall = ''
+    mkdir -p "$out/share/applications"
+    mkdir -p "$out/share/icons/hicolor/48x48/apps"
+    cp transgui.png "$out/share/icons/hicolor/48x48/apps"
+    cp $desktopItem/share/applications/* $out/share/applications
+  '';
+  
+  meta = { 
+    description = "A cross platform front-end for the Transmission Bit-Torrent client";
+    homepage = https://sourceforge.net/p/transgui;
+    license = stdenv.lib.licenses.gpl2Plus;
+    maintainers = with stdenv.lib.maintainers; [ ramkromberg ];
+    platforms = stdenv.lib.platforms.linux;
```

Comment:
Windows and Darwin have build conditions that include different files that need testing on those platforms and can't be assumed to work.
There's also likely issues regarding hard-coded directories that need special attention.